General refactoring and reformatting

LaurenceRawlings opened the pull request:
- use the data class syntax to auto generate constructors
- simplify and move shared `__repr__` and `__eq__` logic to base model class
    - the behaviours implemented are default to data classes (could be removed?)
- added a str function to the base model class to make print output of models prettier
- added `__all__` to main init to allow wildcard imports (`from ergast_py import *`)
- added `black` as a dev dependency and reformatted the project
- fixed styling errors in `README.md`
- marked static functions as static
    - maybe the `Helpers` class does not need to be a class?
- raise value errors instead of returning `None` in some functions to avoid returning an unexpected type defined by the type hint

jamesreprise reviewed (changes requested):
Can we add a CI job to auto-run black? Without this we will likely have regressions.

jamesreprise commented on .gitignore:
```diff
@@ -1,4 +1,5 @@
 .vscode
+.idea
```
```suggestion
```

Let's keep IDE-specific lines out of the codebase. You can ignore these locally in `$XDG_CONFIG_HOME/git/ignore` or `$GIT_DIR/info/exclude`.

jamesreprise commented:
See https://github.com/Samuel-Roach/ergast-py/pull/8 after merging this.

Samuel-Roach commented:
Can this be redirected into the `develop` branch

LaurenceRawlings commented on ergast_py/constants/expected.py:
```diff
@@ -10,8 +11,7 @@ class Expected():
         * "string"
-        * "dict"
-"""
+        * "dict" """
```
Yes this was black, probably can be configured to do it the other way.

LaurenceRawlings commented on ergast_py/models/driver_standing.py:
```diff
@@ -4,10 +4,11 @@
 
 from ergast_py.models.constructor import Constructor
 from ergast_py.models.driver import Driver
+from ergast_py.models.base_model import BaseModel
```
May be worth adding `reorder-python-imports` as a dev dependency and adding it to pre-commit

LaurenceRawlings commented on ergast_py/helpers.py:
```diff
@@ -25,32 +28,37 @@ def construct_datetime_dict(self, datetime_dict: dict) -> datetime.datetime:
-        return None
+        raise ValueError("Time string cannot be empty")
 
     def construct_lap_time(self, time: dict) -> datetime.time:
```
What do you mean sorry?

LaurenceRawlings replied:
What do you mean sorry?

LaurenceRawlings commented on ergast_py/helpers.py:
```diff
@@ -25,32 +28,37 @@ def construct_datetime_dict(self, datetime_dict: dict) -> datetime.datetime:
         Dictionary should contain the keys "date" and "time"
         """
         if "date" not in datetime_dict or "time" not in datetime_dict:
-            return None
+            raise ValueError("Dictionary must contain keys 'date' and 'time'")
         return self.construct_datetime_str(datetime_dict["date"], datetime_dict["time"])
```
I thought this at the time. I think all functions should follow the guard clause flow and raise error on last line.

jamesreprise commented:
<img src="https://media.tenor.com/pFtihBqlOYcAAAAd/kim-kardashian.gif"></img>

LaurenceRawlings commented:
- merged in develop branch to resolve conflicts
- refactored IFs in `helpers.py` to use guard clause style
- added `reorder-python-imports` and `pre-commit` as dev dependencies
- added pre-commit config file to run styling checks
- ran hooks on all files to reorder python imports

jamesreprise commented on .gitignore:
```diff
@@ -1,9 +1,8 @@
 test.py
-*.afphoto
\ No newline at end of file
+*.afphoto
```
Is this necessary?

Samuel-Roach replied:
Ah good spot - It was from previously establishing icons - I guess similarly with `test.py` was initially for local testing and could just be handled with better developer workflow

LaurenceRawlings replied:
Could replace the whole gitignore with one generated by gitignore.io?
https://www.toptal.com/developers/gitignore/api/python

jamesreprise replied:
The guiding principle I personally use is to keep the gitignore restricted to any files we expect to be generated by Python or the tooling that is integrated into the repo (so, if we decide that we're using black we would want to ignore any files that black creates). 

.idea and .vscode don't come under that because they're not repo-specific tooling - developers should be encouraged to add that to their personal, global gitignore. 

All this is to say I agree with your second suggestion.
